license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/AbstractLicenseMojo.java

```diff
@@ -492,13 +500,29 @@ protected final void execute(final Callback callback) throws MojoExecutionExcept
       }
 
       // make default base dir canonical
-      this.defaultBasedir = this.getCanonicalFile(this.defaultBasedir, "license.basedir");
+      workspace.basedir = getCanonicalFile(firstNonNull(workspace.basedir, legacyDefaultBasedir), "license.workspace.basedir");
 
       // collect all the license sets together
       final LicenseSet[] allLicenseSets;
 
       // if we abandon the legacy config this contiguous block can be removed
       final LicenseSet legacyLicenseSet = convertLegacyConfigToLicenseSet();
+
+      if (workspace.basedir != null) {
```

Comment:
this should never happen: the code above ensures `workspace.basedir` will be always set. We could even have a `requireNonNull()` check to make sure of that, to help for tests.